File: BadReadsAPIApp/BadReadsAPI/Book.cs
using System.ComponentModel.DataAnnotations;

namespace BadReadsAPI
{
    public enum BookType
    {
        UNDEFINED,
        Horror,
        Fantasy,
        Thriller,
        Drama,
        Biography,
        Scientific,
        Romance,
        History,
        Poetry,
        Humor,
        Literary,
        Crime
    }
    public class Book
    {
        public int Id { get; set; }
        [StringLength(50)]
        public string Title { get; set; } = string.Empty;
        [StringLength(500)]
        public string Description { get; set; } = string.Empty;
        [StringLength(100)]
        public string Author { get; set; } = string.Empty;
        public float Rating { get; set; }
        public BookType Type { get; set; } = BookType.UNDEFINED;
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: BadReadsUIApp/src/app/models/book.ts
export class Book {

    public id: number;
    public title: string;
    public description: string;
    public author: string;
    public rating: number;
    public type: BookType;

    constructor (
        id: number,
        title: string,
        description: string,
        author: string,
        rating: number,
        type: BookType
    ) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.author = author;
        this.rating = rating;
        this.type = type;
    }

    fromJson(json: any) {
        return new Book(
            json.id,
            json.title,
            json.description,
            json.author,
            json.rating,
            json.type
        );
    }
    
}

export enum BookType {
    UNDEFINED,
    Horror,
    Fantasy,
    Thriller,
    Drama,
    Biography,
    Scientific,
    Romance,
    History,
    Poetry,
    Humor,
    Literary,
    Crime
}
